packages/@react-aria/dnd/src/useDropIndicator.ts

```diff
@@ -62,10 +70,10 @@ export function useDropIndicator(props: DropIndicatorProps, state: DroppableColl
     });
   } else {
     let before = target.dropPosition === 'before'
-      ? collection.getKeyBefore(target.key)
+      ? getKeyBefore(target.key)
```

Comment:
I wonder if `collection.getFirstKey() === target.key` would work? I think TableCollection returns the first row from `getFirstKey()` rather than the header row. Might be able to avoid adding new API that way.

Reply:
My concern with using `getFirstKey`/`getLastKey` is with Sections where the `collection.body.bodyNodes` which `getFirstKey` uses will be the sections themselves rather than the rows. We can go with `getFirstKey()` in this PR for now and I can look into changing that in my PR

Reply:
I think when we have sections we will still want `getFirstKey` to return the first real row rather than the section key. That way tabbing into the table would go to the first row. I think that's how ListCollection works with sections as well?

Reply:
Ya I can make that change in the TableCollection, I was currently handling the tabbing bit via the keyboard delegate logic instead (felt more appropriate to have it there to separate collection vs keyboard specific logic)

Reply:
It turns out that ListCollection actually returns the first section key from `getFirstKey`. I could either change it so all the collection key getters only return row keys (feels like a breaking change?) or add new getters that only return row keys specifically (e.g. getFirstRowKey. getRowKeyAfter). Alternatively, we could provide the `keyboardDelegate`/`layout` to `useDropIndicator` like we do to `useDroppableCollection`, which feels more correct to me.

Reply:
hmm yeah. would have to be optional though because otherwise that would also be a breaking change.